README.md

```diff
@@ -12,31 +12,51 @@ Since the decision to delay standardizing top-level `await` it has come up in a
 
 ## Motivation
 
-The current implementation of `async / await` only support the `await` keyword inside of `async` functions. As such we are beginning to see the following pattern emerge:
+### top-level main
+
+The current implementation of `async / await` only support the `await` keyword inside of `async` functions. As such many programs that are utilizing `await` now make use of top-level main function:
 
 ```js
 import ...
 async function main() {
   const dynamic = await import('./dynamic-thing.mjs');
+  const data = await fetch(dynamic.url);
+  console.log(data);
 }
 main();
 export ...
 ```
 
-This pattern is reminiscent of the classic pattern of wrapping all of your code in a self executing function `(fn(){}())`. This type of "magic" does not benefit new developers and creates excessive repeated code throughout the ecosystem.
+This pattern is creating an abundance of boilerplate code in the ecosystem. If the pattern is repeated throughout the module graph it degrages the determinism of the execution.
 
-Another risk with the above pattern is that it makes the body of the function asynchronous. If people are utilizing this pattern throughout their graph there will no longer be a deterministic execution order.
+### IIAFEs
 
-Another pattern that is beginning to surface is exporting async function and awaiting the results of imports, which drastically impacts our ability to do static analysis of the module graph.
+```mjs
+import static1 from './static1.mjs';
+import { readFile } from 'fs';
 
-```js
-export default async function (url) {
-  if (!fetch) {
-    const fetch = await import('./fetch-polyfill.mjs');
-  }
-  const data = await fetch(url);
-  return data;
-}
+(async () => {
+  const dynamic = await import('./dynamic' + process.env.something + '.js');
+  const file = JSON.parse(await readFile('./config.json'));
+
+  // main program goes here...
+})();
+```
+
+###  completely dynamic modules
+
+```mjs
+export default async () => {
+  // import other modules like this one
+  const import1 = await (await import('./import1.mjs'))();
```

Comment:
missing `.default`

Reply:
can you share the code example you would rather see?

Reply:
`import` returns the full namespaces so the above ~is:

```mjs
const import1Namespace = await import('./import1.mjs');
const import1 = await import1Namespace;
```

So it doesn't actually call / use the `export default` of `import1.mjs`.

```mjs
const import1 = await ((await import('./import1.mjs')).default)();
```

looks like what was intended.